src/steps/owners/index.ts

```diff
@@ -4,26 +4,43 @@ import {
 } from '@jupiterone/integration-sdk-core';
 import { createAPIClient } from '../../client';
 import { IntegrationConfig } from '../../config';
+import { getEntityKey } from '../../utils';
 import { Entities, IntegrationSteps, Relationships } from '../constants';
-import { createOwnerEntity } from './converters';
+import { createOwnerEntity, createOwnerRoleRelationship } from './converters';
 
 export async function fetchOwners({
   jobState,
   instance,
 }: IntegrationStepExecutionContext<IntegrationConfig>) {
   const apiClient = createAPIClient(instance.config);
   await apiClient.iterateOwners(async (owner) => {
-    await jobState.addEntity(createOwnerEntity(owner));
+    // Intermediate step to get owner role through user endpoint
+    const { roleId } = await apiClient.fetchUser(owner.userId.toString());
+
+    const roleEntity = await jobState.findEntity(
+      getEntityKey(Entities.ROLE, roleId),
+    );
+    const ownerEntity = createOwnerEntity(owner);
+    await jobState.addEntity(ownerEntity);
+
+    if (roleEntity && ownerEntity) {
+      await jobState.addRelationship(
+        createOwnerRoleRelationship(ownerEntity, roleEntity),
+      );
+    }
   });
 }
 
 export const ownerSteps: IntegrationStep<IntegrationConfig>[] = [
   {
     id: IntegrationSteps.OWNERS,
     name: 'Fetch Owners',
-    entities: [Entities.USER],
-    relationships: [Relationships.USER_HAS_TEAM],
-    dependsOn: [],
+    entities: [Entities.OWNER],
+    relationships: [
+      Relationships.OWNER_ASSIGNED_ROLE,
+      Relationships.USER_HAS_TEAM,
```

Comment:
We should remove this as this relationship isn't getting created in this step function, but instead in `/steps/teams` (and it's missing there - should be added - I'm aware you didn't write the code there)